src/private_types.h

```diff
@@ -639,7 +639,9 @@ struct ecs_world_t {
     ecs_os_mutex_t sync_mutex;   /* Mutex for job_cond */
     int32_t workers_running;     /* Number of threads running */
     int32_t workers_waiting;     /* Number of workers waiting on sync */
-
+    ecs_os_api_thread_new_t task_worker_new; /* override to create task workers. 0= unused */
+    ecs_os_api_thread_join_t task_worker_join; /* override to join task workers. 0= unused */
```

Comment:
Can the OS API `thread_new` and `thread_join` callbacks be used for this? E.g. you'd set the OS API callbacks to your custom new/join functions first, then enable the task API.

The thread_new and thread_join aren't used for anything else by Flecs except the worker API, so it wouldn't conflict with anything.

Reply:
I had considered this, but the ecs_http_server uses ecs_os_thread_new and I didn't want to change that system. I figured it would be best to maintain the separate APIs for threads and tasks rather than force other systems (or future systems) from needing to be task-compliant.